### DISCUSSION TITLE
Adding Calendar name in ICS file generated via eme_ical_link shotcode

### DISCUSSION THREAD

**Discussion by swissroach** [liedekef/events-made-easy#586] [Q&A] [answered]
Hi Franky, hi everyone

First of all: I love your plugin, best event plugin for WordPress! Thanks a lot for all your time you spend into it!! Our small hobby icehockey club loves it, especially with the queue function for our trainings :)

I would love to additionally set the NAME: and X-WR-CALNAME: at the beginning of the generated .ics file. I saw you set the 4 lines at the beginning of these files in your function eme_ical() in the eme-ical.php file:
BEGIN:VCALENDAR
VERSION:2.0
METHOD:PUBLISH
PRODID:-//hacksw/handcal//NONSGML v1.0//EN

I'm not the pro with Wordpress, so I couldn't figure it out by myself. Can I use some sort of filter or hook to achieve this? Or how should i do this? I don't want to directly edit the eme-ical.php, because that could be replaced with updates of the plugin and it would be simply wrong to do that there ;) I only have one event calendar which gets published as .ics, so it would work for me, but I want to solve this properly :)
Can you give me an advice on how to achieve this?

PS: the X-WR-CALNAME: seems to be used by Exchange Server Protocol, not the simple NAME:, so it would be nice to have both NAME properties in the ics > https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcical/1da58449-b97e-46bd-b018-a1ce576f3e6d

Thanks in advance for any help or tip to my question :)

**Comment by liedekef** [comment, marked as answer]
This should help:
https://github.com/liedekef/events-made-easy/commit/42869bafed7bf29f5222ee4ebd34196778d25b74

After applying this, you can create your own action hook and echo the headers of choice in there.

**Reply by swissroach** [reply]
Awesome, it worked! Thanks a lot!!! Outlook adds the calendar directly with to provided name, very cool.

Just for information to other people how want to use this feature.
I've made an action hook like this (the /r/n at the end of each echoed header is important for the line break in the generated ICS file):

add_action( 'eme_ical_header_action', function () {
   echo "NAME:Calendar Name\r\n";
   echo "X-WR-CALNAME:Calendar Name\r\n";
";
} );

These to NAME headers will have the effect, that the calender will get a proper name when adding it to Outlook. I don't have an iPhone to test it, but I think it will be the same there.

**Reply by liedekef** [reply]
Even better would be to use a function for this:
```
echo eme_esc_ical('NAME:', $cal_name");
echo eme_esc_ical('X-WR-CALNAME:', $cal_name");

```
The function eme_esc_ical adds the needed line endings and takes care of any problematic characters, number of characters etc ...
